pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -131,9 +132,19 @@ rec {
               (lib.concatLists propagatedDependencies));
         in
         {
-          name = name + lib.optionalString
-            (stdenv.hostPlatform != stdenv.buildPlatform)
-            ("-" + stdenv.hostPlatform.config);
+          name =
+            let
+              # Fixed-output derivations like source tarballs shouldn't get a
+              # host suffix. See #32986.
+              isHostSensitive =
+                stdenv.hostPlatform != stdenv.buildPlatform &&
+                (builtins.length buildInputs != 0 ||
```

Comment:
Ah so this part isn't actually true. Say we are building `compiler-rt`; it run-time depends on nothing, but cares about the host platform as much as any other platform.

In that thread, I instead meant that separately we should do a check like:
```nix
let
  fixedOutputDrv = args ? outputHash;
  noNonNativeDeps = builtins.length (depsBuildTarget ++ depsBuildTargetPropagated
                                  ++ depsHostHost ++ depsHostHostPropagated
                                  ++ buildInputs ++ propagatedBuildInputs
                                  ++ depsTargetTarget ++ depsTargetTargetPropagated) == 0;
in
assert fixedOutputDrv -> noNonNativeDeps;
```

Reply:
Technically, even that isn't always *necessarily* true. Yes, any of those deps mean we now need host- or target-platform-specific hashes, but for pre-built stuff (bootstrapping, proprietary, etc) we do just that. So I suppose pre-built stuff could have such deps. But unless all transitive deps were also fixed-output, it would be a total pain to keep the hash correct.

In practice, for the platform-specific deps we do use, run-time deps are hooked up in a downstream deriving, e.g. with `substituteAll` or `makeWrapper`, so I don't there's a practical reason not to have that assertion. But it would be fine to leave it for a separate PR if you like, as it might require fixing miscategorized deps / is a separate issue from this one.